core/headerchain.go

```diff
@@ -1163,6 +1168,41 @@ func (hc *HeaderChain) SlicesRunning() []common.Location {
 	return hc.slicesRunning
 }
 
+func (hc *HeaderChain) ComputeExpansionNumber(parent *types.WorkObject) (uint8, error) {
+	// If the parent is a prime block, prime terminus is the parent hash
+	_, order, err := hc.engine.CalcOrder(hc, parent)
+	if err != nil {
+		return 0, err
+	}
+	var primeTerminusHash common.Hash
+	if order == common.PRIME_CTX {
+		primeTerminusHash = parent.Hash()
+	} else {
+		primeTerminusHash = parent.PrimeTerminusHash()
+	}
+
+	primeTerminus := hc.GetBlockByHash(primeTerminusHash)
+
+	// If the Prime Terminus is genesis the expansion number is the genesis expansion number
+	if hc.IsGenesisHash(primeTerminusHash) && hc.NodeLocation().Equal(common.Location{0, 0}) {
+		return primeTerminus.ExpansionNumber(), nil
+	} else {
```

Comment:
Risk of panic if we don't check primeTerminus != nil
Even though it should not happen here

Reply:
done